### PR TITLE
#166872563 User Should Rate an Order Successfully

### PR DESCRIPTION
**Pull request by Daymorelah**
#### What does this PR do?
Allows a user to rate an order successfully

#### Description of Task to be completed?
* Ensure a user can rate a menu successfully
* Refactor the codebase as see fit
* Ensure all tests pass.

#### How should this be manually tested?
* Clone the repo and `cd` into the project's directory
* Run `npm install` to install dependencies
* Run `npm run build && npm run dev` to start the app
* Navigate to the `/orders` URL
* click on the `order` menu in the side Nav
* Click on the `Rate` button to rate a meal

#### What are the relevant pivotal tracker stories?
[#166872563](https://www.pivotaltracker.com/story/show/166872563)

### CODE REVIEW

File: client/src/components/Admin/Menus/Index.jsx [hunk 1]
```diff
@@ -337,7 +338,7 @@ Menus.propTypes = {
 export const mapStateToProps = (state) => (
   {
     menus: state.menus,
-    upComingEngagements: state.allEngagements.upComingEngagements.engagements
+    upComingEngagements: state.allEngagements.engagements.engagements
```
Comment (azupatrick0) [inline client/src/components/Admin/Menus/Index.jsx]: Hello @Daymorelah please can we destructure the state here

File: client/src/components/common/FormInputs.jsx [hunk 2]
```diff
@@ -21,26 +21,29 @@ const Input = ({
...
+        : ((type !== 'date-picker') && (type !== 'select')
+            && (
+              <input
+                id={id}
+                className="input"
+                name={name}
+                onChange={onChangeHandler}
+                onFocus={clearErrors}
+                value={value}
+                type={type}
```
Comment (zellymiriam) [inline client/src/components/common/FormInputs.jsx]: @Daymorelah The placeholder prop is missing here.

File: client/src/tests/components/OrderHistory/OrderHistory.test.jsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ const menuList = [
       {
         id: 67,
         isDeleted: false,
-        mealType: "side",
+        meal_yype: "side",
```
Comment (zellymiriam) [inline client/src/tests/components/OrderHistory/OrderHistory.test.jsx]: For consistency, I think we should name our variables in camel case.

File: client/src/components/common/FormInputs.jsx (hunk 2 above)
Comment (Daymorelah) [reply client/src/components/common/FormInputs.jsx]: I have added it.

File: client/src/tests/components/OrderHistory/OrderHistory.test.jsx (hunk 3 above)
Comment (Daymorelah) [reply client/src/tests/components/OrderHistory/OrderHistory.test.jsx]: That is what is being sent from the backend. Changing it to camel case causes a bug in the app.

**Pull request merged**